control-plane/agents/src/bin/core/pool/service.rs

```diff
@@ -98,6 +99,15 @@ impl PoolOperations for Service {
             }
             .into());
         }
+
+        if !req.label_key().is_empty() && req.label_key() == dsp_created_by_key() {
+            return Err(SvcError::InvalidLabelCreatedBy {
+                labels: req.label_key(),
+                resource_kind: ResourceKind::Pool,
+            }
+            .into());
+        }
```

Comment:
Actually, these checks should probably be done on the unlabel, rather than at this handler?

Reply:
Yeah, we should do this from operations trait impl.

Reply:
done